waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/mapping/model/MetaStoreMappingFactoryImplTest.java

```diff
@@ -149,4 +155,30 @@ public void loadDefaultMetastoreFilterHook() {
     assertThat(mapping, is(notNullValue()));
     assertThat(mapping.getMetastoreFilter(), instanceOf(DefaultMetaStoreFilterHookImpl.class));
   }
+
+  @Test
+  public void loadMetastoreFilterHookWithCustomConfig() {
+    AbstractMetaStore federatedMetaStore = newFederatedInstance("fed1", thrift.getThriftConnectionUri());
+    federatedMetaStore.setHiveMetastoreFilterHook(PrefixingMetastoreFilter.class.getName());
+    Map<String,String> metaStoreConfigurationProperties = new HashMap<>();
+    metaStoreConfigurationProperties.put(PrefixingMetastoreFilter.PREFIX_KEY,"prefix-test-");
+    federatedMetaStore.setConfigurationProperties(metaStoreConfigurationProperties);
+
+    MetaStoreMapping mapping = factory.newInstance(federatedMetaStore);
+    assertThat(mapping, is(notNullValue()));
+    MetaStoreFilterHook filterHook = mapping.getMetastoreFilter();
+    assertThat(filterHook, instanceOf(PrefixingMetastoreFilter.class));
+
+    Table table = new Table();
+    try {
+      StorageDescriptor sd = new StorageDescriptor();
+      File localWarehouseUri = temporaryFolder.newFolder("local-warehouse");
+      sd.setLocation(new File(localWarehouseUri,"local_database/local_table").toURI().toString());
```

Comment:
Can you please try if a simple string also works, maybe we don't need to create a dummy folder just for string manupilations.:
```sd.setLocation("file://tmp/local_database/local_table");```